File: src/checkers-api/Models/GameModels/Square.cs
namespace checkers_api.Models.GameModels;

public class Square
{
    public Location Location { get; }
    public Color Color { get; }
    public Piece? Piece { get; set; }
    public bool IsOccupied => Piece is not null;

    public Square(Location location, Color color)
    {
        Location = location;
        Color = color;
    }

    public Square(Location location, Color color, Piece piece)
    {
        Location = location;
        Color = color;
        Piece = piece;
    }
}

File: src/checkers-client/src/game-models/square.ts
import Piece from "./piece";

export default class Square {
  public readonly location: Location;
  public readonly color: string;
  public readonly isOccupied: boolean;
  public readonly piece?: Piece;

  constructor(location: Location, color: string, isOccupied: boolean, piece?: Piece) {
    this.location = location;
    this.color = color;
    this.isOccupied = isOccupied;
    this.piece = piece;
  }
}
